src/main.cpp

```diff
@@ -4744,6 +4724,30 @@ PeerLogicValidation::PeerLogicValidation(CConnman* connmanIn) : connman(connmanI
     recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
 }
 
+void PeerLogicValidation::SyncTransaction(const CTransaction& tx, const CBlockIndex* pindex, int nPosInBlock) {
+    LOCK(cs_main);
```

Comment:
@TheBlueMatt Your comment confuses me a bit.  Wouldn't this be broken without that check?